### PR TITLE
feat: client api with proper client segments and strategy variants

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

* Added strategy variants to client API - behind a flag
* Fixing client segments to include only id, constraints and name (optional but I left if for now to ease debugging and we were returning it anyway)

### CODE REVIEW

File: src/lib/db/feature-toggle-client-store.ts [hunk 1]
```diff
@@ -233,15 +241,19 @@ export default class FeatureToggleClientStore
...
+    private rowToStrategy(row: Record<string, any>): IStrategyConfig {
+        const strategy = {
             id: row.strategy_id,
             name: row.strategy_name,
             title: row.strategy_title,
             constraints: row.constraints || [],
             parameters: mapValues(row.parameters || {}, ensureStringValue),
             sortOrder: row.sort_order,
         };
+        if (this.flagResolver.isEnabled('strategyVariant')) {
```
Comment (kwasniew) [inline src/lib/db/feature-toggle-client-store.ts:253]: we can selectively enable it to test different SDKs

File: src/lib/db/segment-store.ts [hunk 2]
```diff
@@ -159,6 +166,20 @@ export default class SegmentStore implements ISegmentStore {
         return rows.map(this.mapRow);
     }
 
+    async getActiveForClient(): Promise<IClientSegment[]> {
```
Comment (kwasniew) [inline src/lib/db/segment-store.ts:169]: same as getActive query but with a different set of fields

File: src/lib/db/feature-toggle-client-store.ts [hunk 3]
```diff
@@ -38,14 +39,22 @@ export default class FeatureToggleClientStore
...
+        flagResolver: IFlagResolver,
+    ) {
         this.db = db;
         this.logger = getLogger('feature-toggle-client-store.ts');
         this.timer = (action) =>
             metricsHelper.wrapTimer(eventBus, DB_TIME, {
                 store: 'feature-toggle',
                 action,
             });
+        this.flagResolver = flagResolver;
```
Comment (kwasniew) [inline src/lib/db/feature-toggle-client-store.ts:57]: we need to control strategy variants at the store query level

File: src/lib/routes/client-api/feature.test.ts [hunk 4]
```diff
@@ -156,13 +158,16 @@ test('fetch single feature', async () => {
...
     });
 
-    return request
-        .get(`${base}/api/client/features/test_`)
-        .expect('Content-Type', /json/)
-        .expect(200)
-        .expect((res) => {
-            expect(res.body.name === 'test_').toBe(true);
-        });
+    return (
```
Comment (sjaanus) [inline src/lib/routes/client-api/feature.test.ts:161]: Are brackets needed?

File: src/lib/routes/client-api/feature.test.ts [hunk 5]
```diff
@@ -156,13 +158,16 @@ test('fetch single feature', async () => {
...
+        request
+            .get(`${base}/api/client/features/test_`)
+            .expect('Content-Type', /json/)
+            // .expect(200)
```
Comment (sjaanus) [inline src/lib/routes/client-api/feature.test.ts:165]: Should comment it in or remove

File: src/lib/routes/client-api/feature.test.ts [hunk 6]
```diff
@@ -156,13 +158,16 @@ test('fetch single feature', async () => {
...
+            .expect((res) => {
+                console.log(res.body);
```
Comment (sjaanus) [inline src/lib/routes/client-api/feature.test.ts:167]: Not needed

File: src/lib/routes/client-api/feature.test.ts [hunk 7]
```diff
@@ -156,13 +158,16 @@ test('fetch single feature', async () => {
...
+                expect(res.body.name === 'test_').toBe(true);
```
Comment (sjaanus) [inline src/lib/routes/client-api/feature.test.ts:168]: Use toEqual instead?

File: src/lib/routes/client-api/feature.test.ts (hunk 4 above)
Comment (kwasniew) [reply src/lib/routes/client-api/feature.test.ts]: some IDE leftovers, fixed now

File: src/lib/routes/client-api/feature.test.ts (hunk 5 above)
Comment (kwasniew) [reply src/lib/routes/client-api/feature.test.ts]: this PR was not ready when you started commenting. It's removed now :)

File: src/lib/routes/client-api/feature.test.ts (hunk 7 above)
Comment (kwasniew) [reply src/lib/routes/client-api/feature.test.ts]: this is old code, not touching it for now

**Review by sjaanus — APPROVED** [review]
LG

**Pull request merged**